oic.r.csr.raml

```diff
@@ -0,0 +1,35 @@
+#%RAML 0.8
+title: OICSecurityCsrResource
+version: v1.0-20150819
+schemas:
+  - Csr: !include schemas/oic.r.csr.json
+
+traits:
+  - interface:
+      queryParameters:
+        if:
+            enum: ["oic.if.baseline"]
+
+/oic/sec/csr:
+  description: |
+    This resource specifies a Certificate Signing Request.
+
+  displayName: Certificate Signing Request
+  is: [ interface ]
+
+  get:
+    description: |
+      Retrieves CSR data.
+    responses:
+      200:
+        body:
+          application/json:
+            schema: Csr
+            example: |
+              {
+              "rt": ["oic.r.csr"],
+              "encoding" : "oic.sec.encoding.pem"
+              "csr": "PEMENCODEDCSR"
+              }
+      503:
```

Comment:
I've heard something about the fixed URI discussion, but frankly, there's a lot of work here and in the code that needs to be done, if we're going to move away from clients using fixed URIs. So I'm not really considering it in this work yet.

Implementing the CSR resource is not optional if the device implements certificates. It's not required to generate a new key pair, but it still must be able to provide a CSR for whatever public key it intends to use.

But yes, if we do move to that, failing to find a resource that implements the csr interface would be the way of doing this, since we would no longer be assuming the resource is kept at a fixed URI. I'm gathering that, when this approach is implemented, querying for the resource types will always return either zero (if not supported/implemented) or one (if it is) instances.